magfest/models.py

```diff
@@ -12,3 +12,11 @@ def age_discount(self):
             if not self.age_group_conf['discount'] or self.age_group_conf['discount'] < half_off:
                 return -half_off
         return -self.age_group_conf['discount']
+
+
+@Session.model_mixin
+class Band:
+    @property
+    def panel_status(self):
+        return str(len(self.group.leader.panel_applications)) + " Panel Application(s)" \
+            if self.group.leader.panel_applications else self.status('panel')
```

Comment:
Shouldn't this go in the `bands` plugin? `bands` is dependent on `panels` so `panel_applications` should _in theory_ always be available from the `bands` plugin.

On the other hand, not every deployment that uses the `magfest` plugin will also have the `bands` and `panels` plugin (like MAGStock). So I think this will break our MAGStock deployment, right?

Reply:
...when is bands dependent on panels?

Very good point about the MAGStock deploy, though. \D:/

Reply:
Our plugin system is not very robust. We have no mechanism for defining explicit dependencies between our plugins. There is, however, a very real implicit dependency tree. Our plugin dependencies manifest in several different places:

1. The `bands` plugin imports the `panels` plugin: https://github.com/magfest/bands/blob/dcd51a600b52476529f7278a3f1dbf158d757cd6/bands/__init__.py#L6
2. The `priority_plugins` setting defines the order in which plugins must be loaded to satisfy the dependencies: https://github.com/magfest/production-config/blob/7b252bf45d2e5eda904b0fc7e0e58bcc11f32fba/event-prime.yaml#L29
3. The `tox` setup in the `bands` plugin installs the `panels` plugin before running the unit tests: https://github.com/magfest/bands/blob/f4816103b8eb9113f6c2e5f4da34fd1785693676/tox.ini#L34
4. And finally, our alembic migration branches reflect the dependency tree:
```
(uber)-+->[uber@head]
       |
       +-(panels)-+->[panels@head]
       |          |
       |          +-(bands)->[bands@head]
       |          |
       |          +-(tabletop)->[tabletop@head]
       |
       +-(attendee_tournaments)->[attendee_tournaments@head]
       |
       +-(hotel)->[hotel@head]
       |
       +-(magprime)->[magprime@head]
       |
       +-(magstock)->[magstock@head]
       |
       +-(mivs)->[mivs@head]
```

Although, I was actually wrong about our MAGStock deployment! We **DO** use the `bands` plugin! We don't have any panels, but MAGStock still includes the `panels` plugin because it's required by the `bands` plugin.

So I think this would work with all of our current deployments, but it will break if we ever create an event without the `bands` or `panels` plugins. Thus, I still believe it belongs in the `bands` plugin 😄

Reply:
Oh wow I had no idea our `bands` plugin imported the `panels` plugin! WHY AM I DOING ALL THIS DECOUPLING

I'll move this to the `bands` plugin after the current outstanding PR there fixed up and merged. :)